### PR TITLE
Add should be credential subject

### PR DESCRIPTION
**Pull request by aljones15**
Addresses: https://github.com/w3c-ccg/vc-api-issuer-test-suite/issues/26

### CODE REVIEW

File: tests/assertions.js [hunk 1]
```diff
@@ -57,3 +54,26 @@ export function shouldBeIssuedVc({issuedVc}) {
...
+    credentialSubject.length.should.be.gt(
+      0,
+      'Expected credentialSubject to make a claim on at least one subject.'
+    );
+    for(const subject of credentialSubject) {
+      subject.should.be.an(
+        'object',
+        'Expected credentialSubject to be an object.'
+      );
+    }
```
Comment (msporny) [inline tests/assertions.js:72]:
```suggestion
    }
```
This should be an object with at least ONE property in it. Ideally, the VC that you pass in should be expanded/recompacted using JSON-LD before sending it to this code to make sure it's a valid JSON-LD object. You could do that in a separate PR.

But one thing you should definitely do is ensure that an empty object isn't provided.

File: tests/assertions.js [hunk 2]
```diff
@@ -57,3 +54,26 @@ export function shouldBeIssuedVc({issuedVc}) {
...
+    return;
+  }
+  credentialSubject.should.be.an(
+    'object',
+    'Expected credentialSubject to be an object.'
+  );
```
Comment (msporny) [inline tests/assertions.js:78]: It needs to not only be an object, but also have at least one internal property.

You should create a function and share it between this code and the code above that checks to see if it's a valid object AND that it has at least one property defined in the object.

File: tests/assertions.js [hunk 3]
```diff
@@ -37,10 +37,7 @@ export function shouldBeIssuedVc({issuedVc}) {
     'Expected `id` to be a string.'
   );
   issuedVc.should.have.property('credentialSubject');
-  issuedVc.credentialSubject.should.be.an(
-    'object',
-    'Expected credentialSubject to be an object.'
-  );
+  _shouldBeCredentialSubject({credentialSubject: issuedVc.credentialSubject});
```
Comment (msporny) [inline tests/assertions.js:40]:
```suggestion
  _shouldBeValidCredentialSubject({credentialSubject: issuedVc.credentialSubject});
```

File: tests/assertions.js [hunk 4]
```diff
@@ -57,3 +54,26 @@ export function shouldBeIssuedVc({issuedVc}) {
       'Expected issuer object to have property id');
   }
 }
+
+function _shouldBeCredentialSubject({credentialSubject}) {
```
Comment (msporny) [inline tests/assertions.js:58]:
```suggestion
function _shouldBeValidCredentialSubject({credentialSubject}) {
```

**Review by msporny — CHANGES_REQUESTED** [review]
The check is only partially correct, we need to check to make sure there is at least one property in each `credentialSubject` as well.

File: tests/assertions.js (hunk 2 above)
Comment (aljones15) [reply tests/assertions.js]: Addressed here: https://github.com/w3c-ccg/vc-api-issuer-test-suite/pull/28/commits/f0878d3f9ea47b76d27f093208d13de0ff216d37

File: tests/assertions.js (hunk 1 above)
Comment (aljones15) [reply tests/assertions.js]: Addressed here: https://github.com/w3c-ccg/vc-api-issuer-test-suite/pull/28/commits/f0878d3f9ea47b76d27f093208d13de0ff216d37

File: tests/assertions.js [hunk 5]
```diff
@@ -57,3 +55,31 @@ export function shouldBeIssuedVc({issuedVc}) {
...
+}
+
+function _shouldHaveClaims({subject}) {
+  subject.should.be.an(
+    'object',
+    'Expected credentialSubject to be an object.'
+  );
+  Object.keys(subject).length.should.be.gt(
+    0,
+    'Expected at least one claim in subject.'
```
Comment (msporny) [inline tests/assertions.js:83]:
```suggestion
    'Expected credentialSubject to have at least one claim.'
```

**Review by msporny — APPROVED** [review]
One minor nit, otherwise LGTM.
